### PR TITLE
[feat] Update The Certificate With Dates From Manifest Header

### PR DESCRIPTION
**Pull request by ajisaxena**
The FMC Alias Certificate was getting populated with default values for "not_before" and "not_after" fields. This PR will let the certificate date fields get populated using the values from manifest.

### CODE REVIEW

File: image/app/src/create/mod.rs [hunk 1]
```diff
@@ -27,6 +28,46 @@ use std::path::PathBuf;
...
+        Ok(d) => d,
+        Err(_) => return Err(anyhow!("Invalid Date Input Format")),
+    };
+
+    if from_date_in < current_date || to_date_in < current_date {
+        return Err(anyhow!("Invalid Input Date"));
+    }
+
+    if from_date_in > to_date_in {
+        return Err(anyhow!("From Date Is Lesser Than To Date"));
```
Comment (mhatrevi) [inline image/app/src/create/mod.rs:65]: This should be "From Date is greater than To Date"

File: image/app/src/create/mod.rs [hunk 2]
```diff
@@ -27,6 +28,46 @@ use std::path::PathBuf;
...
+    if from_date.len() < time_fmt.len() || to_date.len() < time_fmt.len() {
+        return Err(anyhow!("Invalid Date Input Format"));
+    }
+
+    let from_date_str: &str = &from_date[0..8];
+    let to_date_str: &str = &to_date[0..8];
+
+    let from_date_in = match NaiveDate::parse_from_str(from_date_str, "%Y%m%d") {
+        Ok(d) => d,
+        Err(_) => return Err(anyhow!("Invalid Date Input Format")),
```
Comment (mhatrevi) [inline image/app/src/create/mod.rs:52]: Suggest making the error message specific to the error such as "Invalid from date"

File: image/app/src/create/mod.rs [hunk 3]
```diff
@@ -27,6 +28,46 @@ use std::path::PathBuf;
...
+    };
+
+    let to_date_in = match NaiveDate::parse_from_str(to_date_str, "%Y%m%d") {
+        Ok(d) => d,
+        Err(_) => return Err(anyhow!("Invalid Date Input Format")),
```
Comment (mhatrevi) [inline image/app/src/create/mod.rs:57]: Please see comment above regarding specific error messages.

File: image/types/src/lib.rs [hunk 4]
```diff
@@ -213,6 +213,16 @@ pub struct ImagePreamble {
     pub _rsvd: [u32; 2],
 }
 
+#[repr(C)]
+#[derive(AsBytes, FromBytes, Default, Debug)]
+pub struct OwnerSignedData {
+    /// Owner Start Date [ASN1 Time Format] For LDEV-Id certificate: Takes Preference over vendor start dates
+    pub owner_not_before: [u8; 15],
+
+    /// Owner End Date [ASN1 Time Format] For LDEV-Id certificate: Takes Preference over owner end date
```
Comment (mhatrevi) [inline image/types/src/lib.rs:222]: This should be "over vendor end date"

File: image/types/src/lib.rs [hunk 5]
```diff
@@ -213,6 +213,16 @@ pub struct ImagePreamble {
     pub _rsvd: [u32; 2],
 }
 
+#[repr(C)]
+#[derive(AsBytes, FromBytes, Default, Debug)]
+pub struct OwnerSignedData {
+    /// Owner Start Date [ASN1 Time Format] For LDEV-Id certificate: Takes Preference over vendor start dates
```
Comment (mhatrevi) [inline image/types/src/lib.rs:219]: NIT: date

File: rom/dev/src/flow/cold_reset/fmc_alias.rs [hunk 6]
```diff
@@ -358,6 +382,8 @@ impl FmcAliasLayer {
         env: &RomEnv,
         input: &DiceInput,
         output: &DiceOutput,
+        not_before: &[u8; FmcAliasCertTbsParams::NOT_BEFORE_LEN],
```
Comment (mhatrevi) [inline rom/dev/src/flow/cold_reset/fmc_alias.rs:385]: Can we use a single constant?

**Review by mhatrevi — CHANGES_REQUESTED** [review]
Added few comments.

File: image/app/src/create/mod.rs (hunk 1 above)
Comment (ajisaxena) [reply image/app/src/create/mod.rs:65]: From date should be less than to date. It is the start date. So if the from date is greater than to date I am returning error.

File: image/app/src/create/mod.rs (hunk 2 above)
Comment (ajisaxena) [reply image/app/src/create/mod.rs:52]: Will do !

File: image/app/src/create/mod.rs (hunk 3 above)
Comment (ajisaxena) [reply image/app/src/create/mod.rs:57]: Noted!

File: rom/dev/src/flow/cold_reset/fmc_alias.rs (hunk 6 above)
Comment (ajisaxena) [reply rom/dev/src/flow/cold_reset/fmc_alias.rs:385]: These are two constants which are auto generated. One constant is for NOT_BEFORE and another one is for NOT_AFTER. So I am using the respective constants.

File: image/app/src/create/mod.rs (hunk 1 above)
Comment (ajisaxena) [reply image/app/src/create/mod.rs:65]: Oh you are talking about the message. Let me correct that.

File: image/types/src/lib.rs (hunk 4 above)
Comment (ajisaxena) [reply image/types/src/lib.rs:222]: Will take care of this.

**Review by mhatrevi — APPROVED** [review]
Added few comments.

**Pull request merged**